### PR TITLE
Added windows platform support to premake

### PR DESCRIPTION
**Pull request by tavrez**
Win32 and Win64 configuration support for visual studio solutions

### CODE REVIEW

**Comment by tavrez** [comment]
I've signed CLA

**Comment by eustas** [comment]
Sorry for the late response. Lets change the order in the platforms list, so Win64 is listed first (and becomes the default).

**Comment by eustas** [comment]
Also adding "platform" blows the configuration matrix.
Now for linux/gmake there will be:
 * release_win32
 * release_win64
 * debug_win32
 * debug_win64

Sounds ridiculous. So, lets get rid of "win" prefix.

**Comment by eustas** [comment]
There is a nice example how to specify architecture on premake wiki: https://github.com/premake/premake-core/wiki/architecture

**Comment by tavrez** [comment]
I though that `system:"Windows"` in filter configurations will make it only available in windows and results won't change anything in linux/gmake.
Thanks for the response and notes, I'll make it better

File: premake5.lua [hunk 1]
```diff
@@ -13,6 +14,12 @@ filter "configurations:Release"
 filter "configurations:Debug"
   flags { "Symbols" }
 
+filter { "platforms:x64" }
+   architecture "x64"
```
Comment (eustas) [inline premake5.lua]: x86?

File: premake5.lua [hunk 2]
```diff
@@ -13,6 +14,12 @@ filter "configurations:Release"
...
+
+filter { "platforms:x86" }
+   architecture "x86_64"
```
Comment (eustas) [inline premake5.lua]: x86?

**Comment by tavrez** [comment]
Sorry, seems like VS only care about platform name, not what we write in front of architecture. But I fixed mistyping you mentioned.

**Pull request merged**